airflow/providers/docker/operators/docker.py

```diff
@@ -280,9 +279,10 @@ def execute(self, context):
         # Pull the docker image if `force_pull` is set or image does not exist locally
         if self.force_pull or not self.cli.images(name=self.image):
             self.log.info('Pulling docker image %s', self.image)
-            for line in self.cli.pull(self.image, stream=True, decode=True):
-                output = json.loads(line.decode('utf-8').strip())
-                if 'status' in output:
+            for output in self.cli.pull(self.image, stream=True, decode=True):
+                if isinstance(output, str):
+                    self.log.info("%s", output)
+                if isinstance(output, dict) and 'status' in output:
```

Comment:
Looks much better. Is there ever a case where cli.pull will return a str when called with decode=True?

Reply:
@retornam Any comments to the question above?

Reply:
I’m guessing he put it in because the docker api docs don’t state that a dict is returned even though one actually is. This way if they decide to ‘fix’ it, it will still work.

Reply:
The docker api docs aren't clear if it will only return a dict / a string. I added the string check  just in case that happens.

Reply:
According to this: https://docker-py.readthedocs.io/en/stable/api.html#docker.api.image.ImageApiMixin.pull:
> decode (bool) – Decode the JSON data from the server into dicts. Only applies with stream=True

Reply:
@acroos and then right below it is says `(generator or str)` for `return Type`